src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,68 @@
-# TO-DO: Complete the selection_sort() function below 
+# TO-DO: Complete the selection_sort() function below
 def selection_sort( arr ):
     # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
+    for i in range(0, len(arr)):
         cur_index = i
         smallest_index = cur_index
-        # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
+        print("smallest_index: ", smallest_index)
+        print("loop: ", cur_index)
+        # Look for smallest value
+        for j in range(i, len(arr)):
+            print("j: ", j)
+            if arr[j] < arr[smallest_index]:
+                # store smallest value index
+                smallest_index = j
+        #temp hold values
+        smallest_val = arr[smallest_index]
+        cur_val = arr[i]
+        #swap values
+        arr[i] = smallest_val
+        arr[smallest_index] = cur_val
+        print(arr)
+    return arr
 
-        # TO-DO: swap
+print(selection_sort([2, 4, 3, 1]))
 
 
+# def selection_sort(arr):
+#     # loop through n-1 elements
+#     for i in range(1, len(arr)):
+#         cur_index = i
+#         cur_value = arr[i]
+#         prev_index = i - 1
+#         while cur_value < arr[prev_index] and prev_index >= 0:
+#             arr[cur_index] = arr[prev_index]
+#             arr[prev_index] = cur_value
+#             prev_index -= 1
+#             cur_index -= 1
+#     return arr
 
 
-    return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
-def bubble_sort( arr ):
+def bubble_sort(arr):
```

Comment:
nice solution but there might be a few too many IF statements.
consider using a boolean to run your while loop (swapped = True)
initialize the bool to be False at the start and if a swap occurs change it back to True.
ex.
while swapped
   swapped = False
  if swap occured -> swapped = True